code/RedirectedURLHandler.php

```diff
@@ -11,36 +11,36 @@
  * @author scienceninjas@silverstripe.com
  */
 class RedirectedURLHandler extends Extension {
-
 	/**
-	 * Converts an array of key value pairs to lowercase
+	 * Converts the case of the keys of the an array
 	 *
-	 * @param array $vars key value pairs
+	 * @param array $arr Key value pairs
+	 * @param int $c Either CASE_LOWER or CASE_UPPER
 	 * @return array
 	 */
-	protected function arrayToLowercase($vars) {
-		$result = array();
-
-		foreach($vars as $k => $v) {
-			if(is_array($v)) {
-				$result[strtolower($k)] = $this->arrayToLowercase($v);
-			} else {
-			    $result[strtolower($k)] = strtolower($v);
-            }
+	protected function array_change_key_case_unicode($arr, $c = CASE_LOWER) {
+		$c = ($c == CASE_LOWER) ? MB_CASE_LOWER : MB_CASE_UPPER;
+		$ret = array();
+		foreach ($arr as $k => $v) {
+			$ret[mb_convert_case($k, $c, "UTF-8")] = $v;
 		}
-
-		return $result;
+		return $ret;
 	}
 
 	/**
 	 * @throws SS_HTTPResponse_Exception
 	 */
 	public function onBeforeHTTPError404($request) {
-		$base = strtolower($request->getURL());
+		$caseInsensitiveMatching = Config::inst()->get(get_class($this), 'case_insensitive_matching');
```

Comment:
`$this->config()->get('case_insensitive_matching');` is better for accessing the config value

Reply:
As mentioned [here](https://github.com/silverstripe/silverstripe-redirectedurls/pull/50#discussion_r152700826) Extensions do not have a `config()` method